rwengine/src/engine/GameWorld.cpp

```diff
@@ -946,3 +958,53 @@ bool GameWorld::isPaused() const
 	return paused;
 }
 
+VehicleObject* GameWorld::tryToSpawnVehicle(VehicleGenerator& gen)
+{
+	constexpr float kMinClearRadius = 10.f;
+
+	bool clear = true;
+
+	if (gen.remainingSpawns <= 0) {
+		clear = false;
+	}
+
+	if (gen.lastSpawnTime + gen.minDelay > state->basic.timeMS) {
+		clear = false;
+	}
+
+	// Ensure there's no existing vehicles near our spawn point
+	if (clear) {
+		for (auto& v : vehiclePool.objects)
+		{
+			if (glm::distance2(gen.position, v.second->getPosition())
+					< kMinClearRadius * kMinClearRadius) {
+				clear = false;
+				break;
+			}
+		}
+	}
+
+	if (clear) {
+		int id = gen.vehicleID;
+		if (id == -1) {
+			// Random ID found by dice roll
+			id = 134;
+			/// @todo use zone information to decide vehicle id
+		}
+
+		auto vehicle = createVehicle(
+					id,
+					gen.position);
+		vehicle->setHeading(gen.heading);
+		vehicle->setLifetime(GameObject::TrafficLifetime);
+
+		if (gen.remainingSpawns < 101) {
```

Comment:
according to [this, the counter is broken](http://www.gtamodding.com/wiki/014C). It does not mention if a 0 < value < 101 will always, never or randomly spawn though.